common/test/utils.py

```diff
@@ -24,3 +24,24 @@ def write(self, _):
         yield
     finally:
         sys.stderr = savestderr
+
+
+@contextmanager
+def skip_signal(signal, receiver, sender, dispatch_uid=None, weak=False):
+    """
+    ContextManager to skip a signal by disconnecting it, yielding,
+    and then reconnecting the signal.
+    """
+    getattr(signal, "disconnect")(
```

Comment:
There's no need to use the two-argument version of `getattr` here -- `signal.disconnect(...)` would work.